e2e-tests/playwright/e2e/audit-log/LogUtils.ts

```diff
@@ -0,0 +1,144 @@
+   * It compares both primitive and nested object properties.
+   *
+   * @param actual The actual log returned by the system
+   * @param expected The expected log values to validate against
+   */
+  public static validateLog(actual: Log, expected: Partial<Log>) {
+    // Loop through each key in the expected log object
+    Object.keys(expected).forEach(key => {
+      const expectedValue = expected[key as keyof Log];
+      const actualValue = actual[key as keyof Log];
+
+      LogUtils.compareValues(actualValue, expectedValue);
+    });
+  }
+
+  /**
+   * Compare the actual and expected values. Uses 'toBe' for numbers and 'toContain' for strings/arrays.
+   * Handles nested object comparison.
+   *
+   * @param actual The actual value to compare
+   * @param expected The expected value
+   */
+  private static compareValues(actual: any, expected: any) {
+    if (typeof expected === 'object' && expected !== null) {
+      Object.keys(expected).forEach(subKey => {
+        const expectedSubValue = expected[subKey];
+        const actualSubValue = actual?.[subKey];
+        LogUtils.compareValues(actualSubValue, expectedSubValue);
+      });
+    } else if (typeof expected === 'number') {
+      expect(actual).toBe(expected);
+    } else {
+      expect(actual).toContain(expected);
+    }
+  }
+
+  /**
+   * Executes a shell command and returns the output as a promise.
+   *
+   * @param command The shell command to execute
+   * @returns A promise that resolves with the command output
+   */
+  static executeCommand(command: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      exec(
+        command, // NOSONAR: Ignoring the security warning for executing OS commands (typescript:S4721)
+        { encoding: 'utf8', shell: '/bin/bash' },
+        (error, stdout, stderr) => {
+          if (error) {
+            console.error('Error executing command:', error); // NOSONAR
+            reject(`Error: ${error.message}`);
+            return;
+          }
+          if (stderr) {
+            console.warn('stderr warning:', stderr); // NOSONAR
+          }
+          resolve(stdout);
+        },
+      );
+    });
+  }
+
+  /**
+   * Fetches the logs from pods that match the fixed pod selector and applies a grep filter.
+   * The pod selector is:
+   * - app.kubernetes.io/component=backstage
+   * - app.kubernetes.io/instance=redhat-developer-hub
+   * - app.kubernetes.io/name=developer-hub
+   *
+   * @param grepFilter The string to filter the logs using grep
+   * @returns A promise that resolves with the filtered logs
+   */
+  static async getPodLogsWithGrep(grepFilter: string): Promise<string> {
+    const podSelector =
+      'app.kubernetes.io/component=backstage,app.kubernetes.io/instance=rhdh,app.kubernetes.io/name=backstage';
+    const tailNumber = 30;
+    const command = `oc logs -l ${podSelector} --tail=${tailNumber} -c backstage-backend -n ${process.env.NAMESPACE} | grep "${grepFilter}" | head -n 1`;
```

Comment:
@gustavolira you can use existing `NAME_SPACE` variable that is already exported in `env_variables.sh`, so new `NAMESPACE` is not be needed
```suggestion
    const command = `oc logs -l ${podSelector} --tail=${tailNumber} -c backstage-backend -n ${process.env.NAME_SPACE} | grep "${grepFilter}" | head -n 1`;
```